api/src/main/java/com/eleks/academy/pharmagator/scheduler/Scheduler.java

```diff
@@ -14,19 +20,50 @@
 
 @Slf4j
 @Component
-@Profile("!test")
 @RequiredArgsConstructor
 public class Scheduler {
 
     private final List<DataProvider> dataProviderList;
+    private final PriceRepository priceRepository;
+    private final PharmacyRepository pharmacyRepository;
+    private final MedicineRepository medicineRepository;
+    private final ModelMapper modelMapper;
 
-    @Scheduled(fixedDelay = 1, timeUnit = TimeUnit.MINUTES)
+    @Scheduled(fixedDelay = 100, timeUnit = TimeUnit.MINUTES)
     public void schedule() {
         log.info("Scheduler started at {}", Instant.now());
-        dataProviderList.stream().flatMap(DataProvider::loadData).forEach(this::storeToDatabase);
+        dataProviderList.stream()
+                .flatMap(DataProvider::loadData)
+                .forEach(this::storeToDatabase);
+        log.info("Scheduler finished at {}", Instant.now());
     }
 
     private void storeToDatabase(MedicineDto dto) {
-        log.info(dto.getTitle() + " - " + dto.getPrice());
+        Medicine medicine = modelMapper.map(dto, Medicine.class);
+        Price price = modelMapper.map(dto, Price.class);
+
+        Pharmacy pharmacy = new Pharmacy();
+        pharmacy.setName(dto.getPharmacy());
+        Pharmacy pharmacyFromDb = pharmacyRepository.findFirstByName(pharmacy.getName());
+        if (pharmacyFromDb != null) {
+            pharmacy.setId(pharmacyFromDb.getId());
+        }
+        pharmacyRepository.save(pharmacy);
+
+
+        Medicine medFromDb = medicineRepository.findFirstByTitle(medicine.getTitle());
```

Comment:
What if title is doubled for two pharmacies?

Reply:
It's not a problem cause medicines would have one record for two pharmacies.
That table stores just a title

Reply:
That's a problem, because we should have different prices for different pharmacies. The whole point is about price comparison!

Reply:
In that case, when two different pharmacies have the same med,  we store in DB exactly what we want by accessing those meds by id in Medicines and setting needed PharmacyId, price, externalId. These two records would have only the same medId but different data(price pharmacyId externalId).

Reply:
more detailed explanation of my mind I could do in Slack

Reply:
Yep. That's why 'findFirstByTitle' will cause a bug.
It should have additional search parameter.